File: LeetCode-Top100LikedQuestions/1. Two Sum.java
class Solution {
  public static int[] twoSum(int[] nums, int target) {

		int result[] = new int[2];
		Map<Integer, Integer> map = new HashMap<Integer, Integer>();

		for (int i = 0; i < nums.length; i++) {
			int check = target - nums[i];
			int index = i;

			if (map.containsKey(check)) {
				result[0] = i;
				result[1] = (int)map.get(check);
				return result;
			}

			map.put(nums[i], i);
		}

		return result;

	}
}

File: Python/LeetCode-TopInterviewQuestions/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        dict = {};
        
        for i in range(len(nums)):
            dict[nums[i]]= i
            
        #print(dict)
        for i in range(len(nums)):
            if (target - nums[i]) in dict.keys() and dict[target-nums[i]]!=i:
                return [i,dict[target-nums[i]]]
